test/integration/controllers/execute-command.test.js

```diff
@@ -0,0 +1,74 @@
+const supertest = require('supertest');
+const { expect } = require('chai');
+const sinon = require('sinon');
+
+describe('API Execute command', function () {
+  let clock;
+  beforeEach(() => {
+    clock = sinon.useFakeTimers();
+  });
+
+  afterEach(() => {
+    clock.restore();
+  });
+
+  it('works with correct data', async function () {
+    sandbox.stub(sails.helpers.sdtdApi, 'executeConsoleCommand').resolves({ result: 'it worked yay' });
+    const response = await supertest(sails.hooks.http.app).get(`/api/sdtdserver/executeCommand?command=version&serverId=${sails.testServer.id}`);
+    const date = new Date();
+    expect(response.body).to.deep.equal({
+      msg: 'it worked yay',
+      date: date.toISOString(),
+      type: 'commandResponse'
+    });
+    expect(response.statusCode).to.equal(200);
+  });
+
+  it('Handles missing params', async function () {
+    let response = await supertest(sails.hooks.http.app).get(`/api/sdtdserver/executeCommand?serverId=${sails.testServer.id}`);
```

Comment:
It's pretty bad that it responds to a get at all. It would allow someone to post a link, redirect or IMG tag and cause a command to be executed. How hard would it be to make it post only?

Reply:
Very true, shouldn't be too hard to make it a POST.